hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/DataChecksum.java

```diff
@@ -70,6 +74,24 @@ public static Type valueOf(int id) {
     }
   }
 
+  private static final MethodHandle NEW_CRC32C_MH;
+  static {
```

Comment:
I'd like this to all be on demand, so that the introspection is only done if needed, which allows the exception to be thrown at the right point.

Also, could we just back off from failing here at all, just log, track that the operation failed (so no attempt is repeated), and then fall back to the older mechanisms. I don't want things to overreact here

Reply:
Agree, logging would go better. I was trying to avoid silent failure here.